src/mlpack/methods/lsh/lsh_search_impl.hpp

```diff
@@ -362,27 +652,65 @@ void LSHSearch<SortPolicy>::ReturnIndicesFromTable(
 
   // Compute the projection of the query in each table.
   arma::mat allProjInTables(numProj, numTablesToSearch);
+  arma::mat queryCodesNotFloored(numProj, numTablesToSearch);
   for (size_t i = 0; i < numTablesToSearch; i++)
-    //allProjInTables.unsafe_col(i) = projections[i].t() * queryPoint;
-    allProjInTables.unsafe_col(i) = projections.slice(i).t() * queryPoint;
-  allProjInTables += offsets.cols(0, numTablesToSearch - 1);
-  allProjInTables /= hashWidth;
+    queryCodesNotFloored.unsafe_col(i) = projections.slice(i).t() * queryPoint;
+  queryCodesNotFloored += offsets.cols(0, numTablesToSearch - 1);
+  allProjInTables = arma::floor(queryCodesNotFloored/hashWidth);
 
   // Compute the hash value of each key of the query into a bucket of the
   // 'secondHashTable' using the 'secondHashWeights'.
-  arma::rowvec hashVec = secondHashWeights.t() * arma::floor(allProjInTables);
+  arma::rowvec hashVec = secondHashWeights.t() * allProjInTables;
 
+  // mod and floor hashVec to compute 2nd-level codes
   for (size_t i = 0; i < hashVec.n_elem; i++)
     hashVec[i] = (double) ((size_t) hashVec[i] % secondHashSize);
 
   Log::Assert(hashVec.n_elem == numTablesToSearch);
 
+  // Compute hashVectors of additional probing bins
+  arma::mat hashMat;
```

Comment:
**On your first point:** I wondered too when I first saw this - I thought it had something to do with operator precedence but [typecasting precedes modding](http://en.cppreference.com/w/c/language/operator_precedence), so I'm fixing that.

**On your second point:** In the beginning I started doing it like that to avoid the copy, but the armadillo spans confused me so I went around it. I'll try to find a way to do it the way you say.